src/py42/services/_connection.py

```diff
@@ -247,7 +248,13 @@ def _handle_error(method, url, response):
     try:
         response.raise_for_status()
     except HTTPError as ex:
-        raise_py42_error(ex)
+        if response.json()["error"][0]["primaryErrorKey"] in [
```

Comment:
the whole if block could be added to `raise_py42_error` so that it continues to be the main function for determining what error should be raised

Reply:
Moved everything over there in 3513cf0